### PR TITLE
Add predict_quantiles to FBprophet

### PR DESCRIPTION
**Pull request by kejsitake**
#### Reference Issues/PRs

See https://github.com/alan-turing-institute/sktime/issues/1687

#### What does this implement/fix? Explain your changes.

- Refactors `predict_quantiles` for FBProphet.
- Changes the test to only check if `pred_int['upper'] >= pred_int['lower']`.
- Implements a fix for when the X contains a variable named "ds", by copying the column to another column and renaming in the case the column is numeric and not an index. This is done in order to be more consistent with the other forecasters, which don't throw an error if exogenous X contains a column "ds" related to #1712.

### CODE REVIEW

**Comment by fkiraly** [comment]
@kejsitake, might the defaults implemented in https://github.com/alan-turing-institute/sktime/pull/1879 perhaps help with this?
It would allow you to implement `_predict_interval` only.

**Comment by kejsitake** [comment]
> @kejsitake, might the defaults implemented in #1879 perhaps help with this? It would allow you to implement `_predict_interval` only.

Thank you. I will explore implementing just `predict_interval`. I suspect that the issues I am encountering are due to the additional index ("ds") that Prophet generates and requires.

**Comment by fkiraly** [comment]
fron our discussion - looks like this failed `assert np.all(pred_int["upper"] > y_pred)`.
Generally, the predictive mean (`y_pred`) does not need to have a relation with any quantile (`pred_int["upper"]`), so that is not a correct check on the output. So, occasionally it can happen that this fails.

The fact that the tests run locally for you and me, I'd put down to random number generator behaviour.

In short: looks like the test is broken, not the estimator.

I'd just change the two lines 276, 277 of the test in question to
`assert np.all(pred_int["upper"] >= pred_int["lower"])`

**Comment by fkiraly** [comment]
& looks like you already fixed it!

File: sktime/forecasting/base/adapters/_fbprophet.py [hunk 1]
```diff
@@ -76,21 +75,17 @@ def _fit(self, y, X=None, fh=None, **fit_params):
...
+    def _predict(self, fh=None, X=None):
         """Forecast time series at future horizon.
 
         Parameters
         ----------
         fh : array-like
             The forecasters horizon with the steps ahead to to predict.
             Default is
-            one-step ahead forecast, i.e. np.array([1]).
+            one-step ahead forecast, i.e. np.arraygit r([1]).
```
Comment (fkiraly) [inline sktime/forecasting/base/adapters/_fbprophet.py:86]: typo

File: sktime/forecasting/base/adapters/_fbprophet.py [hunk 2]
```diff
@@ -114,19 +109,86 @@ def _predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
...
+
+        Returns
+        -------
+        quantiles : pd.DataFrame
+            Column has multi-index: first level is variable name from y in fit,
+                second level being the values of alpha passed to the function.
+            Row index is fh. Entries are quantile forecasts, for var in col index,
+                at quantile probability in second col index, for the row index.
+        """
+        self.check_is_fitted()
```
Comment (fkiraly) [inline sktime/forecasting/base/adapters/_fbprophet.py:148]: these two checks are already done in the "non-underscore" function, kindly remove

File: sktime/forecasting/tests/test_all_forecasters.py [hunk 3]
```diff
@@ -327,6 +327,7 @@ def test_predict_pred_interval(Forecaster, fh, alpha):
 def _check_predict_quantiles(
     pred_quantiles: pd.DataFrame, y_train: pd.Series, fh, alpha
 ):
+
```
Comment (fkiraly) [inline sktime/forecasting/tests/test_all_forecasters.py:330]: typo, remove empty line before function docstring

**Comment by fkiraly** [comment]
Quick question: have you checked that it's fine to overwrite the `inverval_width` late? Do the quantiles look sensible?

**Comment by kejsitake** [comment]
Thank you for the comments!

> Quick question: have you checked that it's fine to overwrite the `inverval_width` late? Do the quantiles look sensible, i.e., that you get similar answers when you change it early, compared to changing it late?

I am not sure what do you mean by early and late. Do you mean when creating object vs. changing the private variable? If yes, I tried printing the output and it looked reasonable.

**Comment by fkiraly** [comment]
> Do you mean when creating object vs. changing the private variable? If yes, I tried printing the output and it looked reasonable.

Yes, that's exactly what I meant, early = at construction, late = before invoking prediction. Sorry for being unclear.
If it looked reasonable/similar, probably all is fine...

File: sktime/forecasting/base/adapters/_fbprophet.py (hunk 1 above)
Comment (kejsitake) [reply sktime/forecasting/base/adapters/_fbprophet.py]: Removed!

**Review by fkiraly — APPROVED** [review]
Thanks, looks good!

Tests ran and passed, I just accidentally restarted them.
